src/components/modal/BorrowDocumentModal.tsx

```diff
@@ -0,0 +1,191 @@
+import { CreateNewFolderOutlined } from '@mui/icons-material'
+import { Box, Button, FormControl, Grid, TextField, Typography } from '@mui/material'
+import { useFormik } from 'formik'
+import * as yup from 'yup'
+import { notifySuccess } from '~/global/toastify'
+import ModalLayout from './ModalLayout'
+import { BorrowRequest } from '~/global/interface'
+import useBorrowRequestApi from '~/hooks/api/useBorrowRequestApi'
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs'
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider'
+import { DatePicker } from '@mui/x-date-pickers/DatePicker'
+
+interface BorrowDocumentModalProps {
+  open: boolean
+  fileId: string
+  fileName: string
+  handleClose: () => void
+}
+
+const BorrowDocumentModal = (props: BorrowDocumentModalProps) => {
+  const { open, handleClose, fileId, fileName } = props
+  const { createBorrowRequest } = useBorrowRequestApi()
+  const validationSchema = yup.object({
+    description: yup.string().required('Description is required').trim(),
+    borrowDuration: yup
+      .number()
+      .required('Borrow duration is required')
+      .min(1, 'Borrow duration must be at least 1 day')
+      .max(30, 'Borrow duration must be at most 30 days')
+  })
+
+  const formik = useFormik({
+    initialValues: {
+      document: {
+        id: fileId
+      },
+      description: '',
+      borrowDuration: 1,
+      startDate: new Date()
+    },
+    validationSchema: validationSchema,
+    onSubmit: async (values: BorrowRequest, { resetForm }) => {
+      try {
+        await createBorrowRequest(values)
+        notifySuccess('Borrow document sent successfully')
+        resetForm()
+        handleClose()
+      } catch (error) {
+        console.log(error)
+      }
+    }
+  })
+
+  const shouldDisableDate = (date: Date) => {
+    return date < new Date()
+  }
+
+  return (
+    <ModalLayout open={open} handleClose={handleClose}>
+      <Box
+        display={'inline-flex'}
+        sx={{
+          p: {
+            xs: 1.5,
+            sm: 3
+          },
+          position: 'sticky',
+          top: 0,
+          background: 'white',
+          zIndex: 1,
+          width: '100%',
+          boxShadow: 'rgba(100, 100, 111, 0.2) 0px 7px 29px 0px'
+        }}
+      >
+        <CreateNewFolderOutlined fontSize='large' sx={{ color: 'var(--black-color)', mx: 1 }} />
```

Comment:
Should change new icon:v. Recommend: import EventIcon from '@mui/icons-material/Event';